pkg/ksl/compiler.go

```diff
@@ -28,7 +40,22 @@ func Compile(r io.Reader) (*intermediate.Namespace, error) {
 	tokens := antlr.NewCommonTokenStream(lexer, antlr.LexerDefaultTokenChannel)
 	interpreter := parser.NewkslParser(tokens)
 
+	errorListener := &CustomErrorListener{}
+	interpreter.RemoveErrorListeners()
+	interpreter.AddErrorListener(errorListener)
+
 	file := interpreter.File()
+
+	if len(errorListener.Errors) > 0 {
+		fmt.Println("Parsing errors detected:")
```

Comment:
We'll want to return an error here and let the caller decide what to do. Panics will halt the program by default or potentially bubble up to a top level error handler, neither of which will work in a loop.

We also most likely want the error detail in a programmatically digestible form (line/col/message) so the caller can decide how to represent it. For example: the transpiler will want to print human-readable messages and exit with a non-zero status code, while the language server we're starting to develop would want to return the details to its caller to put error highlights on an IDE screen and keep running.

It's also not typical to print anything on success.